### PR TITLE
Restore old CLI `info` command output schema with federation details

### LINKED ISSUE #2973: Regressing in `fedimint-cli info`
When switching to the new client we didn't notice [the regression in the info command output](https://discord.com/channels/990354215060795454/990354215878688860/1141552028452126861). We still have the old schema in our docs and should restore it:

https://github.com/fedimint/fedimint/blob/7123a025343c025f589a33e44ddd4bc601c6d8e1/docs/dev-running.md?plain=1#L28-L43

### PR DESCRIPTION
**Pull request by shaurya947**
Also adds `preserve_order` feature for `serde_json` to ensure keys appear in the same order in the output as they are defined in the code.

Resolves #2973

### CODE REVIEW

**Review by elsirion — COMMENTED** [review]
Fixing the regression for me is more about having the same data available again. But naming improvements are good imo. If you rename something you'll have to also change it in devimint, e.g. currently CI fails here:

https://github.com/fedimint/fedimint/blob/1de431a70fb1122a9b3210500f2f00f7a62677c5/devimint/src/federation.rs#L289-L293

**Comment by elsirion** [comment]
PR/commit style is great btw!

File: fedimint-cli/src/client.rs [hunk 1]
```diff
@@ -452,20 +453,30 @@ pub async fn handle_ng_command(
...
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 struct InfoResponse {
-    total_msat: Amount,
-    denominations_msat: TieredSummary,
+    federation_id: FederationId,
+    network: Network,
+    meta: BTreeMap<String, String>,
+    total_amount: Amount,
```
[the remark this answers is not in the document]
Comment (shaurya947) [reply fedimint-cli/src/client.rs:477]: Agreed, I'll use `total_amount_msat` and also keep `denominations_msat`.

**Comment by shaurya947** [comment]
Updated! Thanks for the `devimint` pointer, it seems to be a sort of harness for integration-like tests—is that correct? How should I think about `devimint` in comparison to the `integrationtests` package?

**Comment by elsirion** [comment]
`integrationtests` runs the federation and client in one process and has the ability to mock external services (e.g. `bitcoind`). `devimint` sets up a federation in a more realistic way by running separate daemons etc. but loses the ability to use mocks. So overall I'd say `integrationtests` is more fine-grained while `devimint` really only covers the basics.

File: fedimint-cli/Cargo.toml [hunk 2]
```diff
@@ -40,7 +40,7 @@ thiserror = "1.0.39"
 tokio = { version = "1.26.0", features = ["full"] }
 tracing ="0.1.37"
 tracing-subscriber = { version = "0.3.16", features = [ "env-filter" ] }
-serde_json = "1.0.91"
+serde_json = { version = "1.0.91", features = ["preserve_order"] }
```
Comment (elsirion) [inline fedimint-cli/Cargo.toml:43]: Not sure if strictly necessary, but I like it.

**Review by elsirion — APPROVED** [review]
Thx for the PR!

**Pull request merged**